ch_pipeline/analysis/timing.py

```diff
@@ -83,6 +132,34 @@ def process(self, tstream):
         # If requested, extract the input flags
         input_flags = tstream.input_flags[:] if self.use_input_flags else None
 
+        # Check needs_timing_correction flags time ranges to see if input timestream
+        # falls within the interval
+        needs_timing_correction = False
+        for flag in self.flags:
+            if (
+                timestamp[0] >= flag["start_time"]
+                and timestamp[0] <= flag["finish_time"]
+            ):
+                if timestamp[-1] >= flag["finish_time"]:
+                    raise PipelineRuntimeError(
+                        f"Sample covering {timestamp[0]} to {timestamp[-1]} partially overlaps "
+                        f"needs_timing_correction DataFlag covering {ephemeris.unix_to_datetime(flag['start_time']).strftime('%Y%m%dT%H%M%SZ')} "
+                        f"to {ephemeris.unix_to_datetime(flag['finish_time']).strftime('%Y%m%dT%H%M%SZ')}."
+                    )
+                else:
+                    self.log.info(
+                        f"Sample covering {timestamp[0]} to {timestamp[-1]} flagged by "
+                        f"needs_timing_correction DataFlag covering {ephemeris.unix_to_datetime(flag['start_time']).strftime('%Y%m%dT%H%M%SZ')} "
+                        f"to {ephemeris.unix_to_datetime(flag['finish_time']).strftime('%Y%m%dT%H%M%SZ')}. Timing correction will be applied."
+                    )
+                    needs_timing_correction = True
+
+        if not needs_timing_correction:
+            self.log.info(
+                f"Sample in span {timestamp[0]} to {timestamp[-1]} does not need timing correction"
```

Comment:
Probably call it data rather than sample.

Also, if you're giving formatted datetimes elsewhere you should probably give them here too.